### ISSUE TITLE
[BUG] After merger of GoPro MAX files, sometimes a part of the video becomes jumpy

### ISSUE THREAD

**Issue by ydutertre** [rubegartor/ReelSteady-Joiner#73]
**Describe the bug**
After merger of GoPro MAX files, sometimes a part of the video becomes jumpy. Each individual GoPro MAX file is fine, playback is very smooth. But after merging via Reelsteady, sometimes (not always) part of the video (usually the part corresponding to the last clip being merged) will become jumpy, with the FOV bouncing abruptly every few seconds.

**To Reproduce**
- Merge multiple GoPro MAX files (as attached)
- Check the created file, especially the portion corresponding to the last clip

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows
 - Version 22H2
 - ReelSteady Joiner Version 1.3.2

**GoPro camera and files information**
(If applicable)
- GoPro Model: GoPro MAX
- GoPro Files info: .360 files at 5.6K, 30fps

**Additional context**

**[IMPORTANT] Video Files**

https://drive.google.com/drive/folders/1PGT0yrohWRx6xLxRgya_uBG_jFOt_20d?usp=sharing
(files being uploaded, will be ready a couple of hours after posting)
After joining these files you can look at around 1:15:30 (or any other time after that or within 9 minutes 28 seconds of the end) to see the footage jumps every few seconds. This seems to correspond to the last clip joined (the clip itself doesn't have this jumpiness).

**[IMPORTANT] 

[GS010094_Jan 19, 2024 6-57 PM.log](https://github.com/rubegartor/ReelSteady-Joiner/files/13987715/GS010094_Jan.19.2024.6-57.PM.log)

**Comment by DennisOtto-dev** [comment]
is there any quickfix on how to avoid that?

**Comment by rubegartor** [comment]
I'm reviewing it, during the next few days I hope to be able to tell you something more.

**Comment by DennisOtto-dev** [comment]
I experienced this issue starting with the second clip. the first clip (8:02min) long is fine and right when the second clip begins there is a strong shaking in the video until the end of all clips.

**Comment by rubegartor** [comment]
Unfortunately no, I have not yet been able to identify the problem. If you could share your clips it would be a big help in trying to find the problem.

**Comment by DennisOtto-dev** [comment]
sorry, I would love to help, but these are private family vacation recordings

**Comment by ydutertre** [comment]
Thanks for looking at this! Please let me know when it would be OK for me to delete the clips from cloud storage, as they are eating up my quota :-)

No urgency on my part, for now I'm manually converting the clips to MP4 then joining in DaVinci Resolve before injecting back 360 metadata, seems to work fine as a workaround

**Comment by rubegartor** [comment]
You can delete the clips without any problem!

**Comment by ydutertre** [comment]
Weird, I don't have that issue. I can load up just fine to 24 minutes
something... (Full length of each sub recording)

**Comment by rubegartor** [comment]
@DennisOtto-dev I think I have it, throughout these days I hope to be able to share a pre-release so you can carry out your own tests

**Comment by DennisOtto-dev** [comment]
I am also excited! Looking forward to the pre-release

**Comment by ydutertre** [comment]
Awesome! Looking forward to it!

**Comment by rubegartor** [comment]
I have just released a pre-release with new changes that should solve this problem, there is a new option: Processing type, I recommend using it with the value: MP4Merge, carry out your own tests, I hope this will solve the problems.

[ReelSteady Joiner 1.3.3-beta](https://github.com/rubegartor/ReelSteady-Joiner/releases/tag/1.3.3-beta)

It would be helpful to know the test results!

**Comment by mglakner** [comment]
I had the same issue, just tried the beta 1.3.3 release and can confirm the jumpiness is gone. Thank you!

**Comment by rubegartor** [comment]
> I had the same issue, just tried the beta 1.3.3 release and can confirm the jumpiness is gone. Thank you! on Macbook Pro M1 Pro.
> 
> Tried on an intel mac mini and got an "Unhandled Error"
> 
> Error: spawn /Applications/ReelSteady Joiner.app/Contents/app/bin/mac/mp4merge EACCES at ChildProcess._handle.onexit (node:internal/child_process:284:19) at onErrorNT (node:internal/child_process:477:16) at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

I have updated the installers and the problem should disappear. If you can confirm it, I would appreciate it since I do not have a Mac with a x64 processor.

**Comment by mglakner** [comment]
> > I had the same issue, just tried the beta 1.3.3 release and can confirm the jumpiness is gone. Thank you! on Macbook Pro M1 Pro.
> > Tried on an intel mac mini and got an "Unhandled Error"
> > Error: spawn /Applications/ReelSteady Joiner.app/Contents/app/bin/mac/mp4merge EACCES at ChildProcess._handle.onexit (node:internal/child_process:284:19) at onErrorNT (node:internal/child_process:477:16) at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
> 
> I have updated the installers and the problem should disappear. If you can confirm it, I would appreciate it since I do not have a Mac with a x64 processor.

Can confirm, 1.3.3 Beta works on x64 and Apple Silicon.

**Comment by DennisOtto-dev** [comment]
I still have the same problem with 1.3.3 beta. It's better, but I still have shaking in my video clip. I have merged six 360 videos from the GoPro MAX together (total length: 40min) and experience shaking throughout the video after the second clip begins.

**Comment by ydutertre** [comment]
Apologies for the late reply, I was waiting to have some long video to test it out. This seems to have fixed the issue for me! I just merged 7*24 minute long GoPro MAX clips, and I don't see any issue. Note however that I am using the Beta firmware of GoPro Max that allows longer clips, so maybe related?

@DennisOtto-dev you are absolutely sure you were running 1.3.3 beta and had restarted the PC first (I needed to do that)
